### PR TITLE
[FLOSS T1] Add exception tests for key c++ bindings

### PR DESCRIPTION
**Pull request by sedadas**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

## Notes
Missing tests found by using the code coverage reports:
https://doc.libelektra.org/coverage/master/debian-bullseye-full/src/bindings/cpp/include/key.hpp.gcov.html
L892, L906, L932

### CODE REVIEW

File: src/bindings/cpp/tests/testcpp_key.cpp [hunk 1]
```diff
@@ -351,6 +351,21 @@ TEST (key, exceptions)
...
 		succeed_if (test.getName () == "user:/name", "name not unchanged");
 	}
+
+	try
+	{
+		test.addName ("no");
+	}
+	catch (kdb::KeyInvalidName const &)
+	{
+		succeed_if (test.getName () == "user:/name", "name not unchanged");
```
Comment (markus2330) [inline src/bindings/cpp/tests/testcpp_key.cpp:361]: The error message is actually quite unhelpful:

```suggestion
		succeed_if (test.getName () == "user:/name", "name not unchanged after trying to add name to invalid named key");
```

Please also correct other unhelpful messages above.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Is there some overlap with the PR #4169? The features there could be tested like this, too.

**Comment by sedadas** [comment]
> Is there some overlap with the PR #4169? The features there could be tested like this, too.

I think that there is. Should we merge this first and then work in the other PR or should we do something else?

**Comment by markus2330** [comment]
First come, first served: if you get the PR ready sooner we can merge this. Please warn @flo91 how his PR might get affected.

**Comment by flo91** [comment]
[FLOSS T1] - Review

Thanks for the PR!
The first line in the commit messages should have the format `module: short statement`.
Your commit messages are missing to `module:` part.
A reference to an issue is not in the description of the PR or in the commit messages, but I don't know how important this criteria is for our FLOSS-PRs because most PRs we created until now don't have a direct relation to an issue.

The buildserver-checks have all passed successfully, congratulations!
According to GitHub, your branch `add-key-cpp-bindings-exception-tests` is 7 commits behind `ElektraInitiative:master`, so rebasing is still necessary.


**About the code:**
I already added  a posting to your PR in GitHub about the misleading code-comments that mention "null keys".
Therefore, I think our test cases don't really overlap, because you use keys that were created with the default constructor, while I use null-keys (Key variable set to `nullptr`) and keys initialized with custom values to test that functions.

I changed and extended my PR #4169 greatly since the discussion about our overlapping PRs was started, so that should probably be reviewed again.

**Comment by markus2330** [comment]
Thank you for the test! :sparkler:

**Pull request merged**